lib/Dialect/Arc/Transforms/InferMemories.cpp

```diff
@@ -98,11 +98,15 @@ void InferMemoriesPass::runOnOperation() {
         params.getAs<IntegerAttr>("numReadPorts").getValue().getZExtValue();
     for (unsigned portIdx = 0; portIdx != numReadPorts; ++portIdx) {
       auto address = instOp.getOperand(argIdx++);
-      auto enable = instOp.getOperand(argIdx++);
+      ++argIdx; // skip enable argument
       auto clock = instOp.getOperand(argIdx++);
       auto data = instOp.getResult(resultIdx++);
-      Value readOp = builder.create<MemoryReadPortOp>(wordType, memOp, address,
-                                                      clock, enable);
+      // NOTE: the result of a disabled read port is undefined, currently we
+      // define it to be the same as if it was enabled, but we could also set it
+      // to any constant (e.g., by inserting a mux).
+      Value readOp = builder.create<MemoryReadPortOp>(wordType, memOp, address);
+      // NOTE: if the read-latency is 0, the memory read is combinatorial
+      // without any buffer
```

Comment:
It's nice that we weren't really using the clock for anything already. Dropping the clock and enable looks like it was more of a "remove unused stuff" than "rework because it changes semantics". Very handy!

Reply:
Yes, it was indeed very convenient that the pass already inserted registers for higher latency reads!